src/frontend/src/btc/schedulers/btc-wallet.scheduler.ts

```diff
@@ -45,16 +65,30 @@ export class BtcWalletScheduler implements Scheduler<PostMessageDataRequestBtc>
 			network: bitcoinNetwork
 		});
 
-		// TODO: Fetch and parse transactions
-		const transactions: BitcoinTransaction[] = [];
+		let uncertifiedTransactions: CertifiedData<BitcoinTransaction>[] = [];
+
+		if (data?.shouldFetchTransactions) {
+			const btcAddress =
+				data?.btcAddress ??
+				this.btcAddress ??
+				(await this.loadBtcAddress({ identity, bitcoinNetwork }));
+
+			const transactions = (await btcAddressData({ btcAddress })).txs;
```

Comment:
- nitpick to align our code style: `const {txs: transactions} = ...`

- `btcAddressData` returns the transactions? isn't the naming of the function a bit misleading?

Reply:
* Ok
* that function returns not only txs data, but for now and in this particular case, I only need `btcAddressData.txs`. If by the end of the implementation we end up only needing transactions info, I can refactor this method to ignore the rest of the data and rename it accordingly.

Reply:
Sounds good